src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -243,6 +248,40 @@ public override async Task<CompletionChange> GetChangeAsync(Document document, C
             return CompletionChange.Create(textChange);
         }
 
+        private static async Task<TextSpan> GetTextChangeSpanAsync(Document document, TextSpan startSpan, CancellationToken cancellationToken)
+        {
+            var result = startSpan;
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(result.Start);
+            if (syntaxFacts.IsStringLiteral(token) || syntaxFacts.IsVerbatimStringLiteral(token))
+            {
+                result = token.Span;
+                var contentStartsAt = token.Text.IndexOf('"');
+                contentStartsAt = contentStartsAt >= 0
+                    ? contentStartsAt + 1
+                    : contentStartsAt;
+                var contentEndsAt = token.Span.Length == 0
+                    ? token.Span.End
+                    : token.Text[token.Span.Length - 1] == '"'
+                        ? token.Span.End - 1
+                        : await GetEndOfOpenEndedStringLiteral(startSpan.End).ConfigureAwait(false); // The string is open ended. We keep the right end as is.
+                result = TextSpan.FromBounds(
+                    start: token.Span.Start + contentStartsAt,
+                    end: contentEndsAt);
+            }
+
+            return result;
+
+            async Task<int> GetEndOfOpenEndedStringLiteral(int startPosition)
+            {
+                var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                var span = CommonCompletionUtilities.GetWordSpan(text, startPosition, ch => false, ch => !char.IsWhiteSpace(ch), alwaysExtendEndSpan: true);
```

Comment:
basically, you could consume until you hit a quote or newline char.  that would consume the right amount in both directions.

Reply:
Except escaped double quotes. That's why I used the approach with the StringLiteralToken. But I'm with you: Escaped double quotes are highly unlikely in this context (and maybe even illegal).

I will give it a try later as it should indeed be much simpler code.

Reply:
👍 .  I'm 100% ok with this not doing a perfect job in such esoteric cases.